tests/unit/test_library_saml.py

```diff
@@ -155,3 +185,44 @@ def test_consumer_charm_emits_event_when_leader():
     assert harness.charm.events[0].metadata_url == relation_data["metadata_url"]
     assert harness.charm.events[0].certificates == tuple(relation_data["x509certs"].split(","))
     assert harness.charm.events[0].endpoints == (slo_ep, sso_ep)
+
+
+def test_privider_charm_doesnt_emit_duplicates():
+    """
+    arrange: set up a charm and set the relation data.
+    act: trigger the update of the relation data again.
```

Comment:
Why the update relation is emitted multiple times?

Reply:
It's a bit unclear. Changed